src/main/java/org/zendesk/client/v2/model/Trigger.java

```diff
@@ -0,0 +1,248 @@
+
+      public List<Condition> getAll() {
+         return all;
+      }
+
+      public void setAll(List<Condition> all) {
+         this.all = all;
+      }
+
+      public List<Condition> getAny() {
+         return any;
+      }
+
+      public void setAny(List<Condition> any) {
+         this.any = any;
+      }
+
+      @Override
+      public String toString() {
+         final StringBuilder sb = new StringBuilder();
+         sb.append("Conditions");
+         sb.append("{all=").append(all);
+         sb.append(", any=").append(any);
+         sb.append('}');
+         return sb.toString();
+      }
+   }
+
+   public static class Condition {
+      private String field;
+      private String operator;
+      private String value;
+
+      public Condition() {}
+
+      public Condition(String field, String operator, String value) {
+         this.field = field;
+         this.operator = operator;
+         this.value = value;
+      }
+
+      public String getField() {
+         return field;
+      }
+
+      public void setField(String field) {
+         this.field = field;
+      }
+
+      public String getOperator() {
+         return operator;
+      }
+
+      public void setOperator(String operator) {
+         this.operator = operator;
+      }
+
+      public String getValue() {
+         return value;
+      }
+
+      public void setValue(String value) {
+         this.value = value;
+      }
+
+      @Override
+      public String toString() {
+         final StringBuilder sb = new StringBuilder();
+         sb.append("Condition");
+         sb.append("{field=").append(field);
+         sb.append(", operator=").append(operator);
+         sb.append(", value=").append(value);
+         sb.append('}');
+         return sb.toString();
+      }
+
+   }
+
+   @JsonDeserialize(using = ActionDeserializer.class)
+   public static class Action {
+      private String       field;
+      private List<String> value;
+
+      public Action() {}
+
+      public Action(String field, String... values) {
+         this.field = field;
+         this.value = Arrays.asList(values);
+      }
+
+      public String getField() {
+         return field;
+      }
+
+      public void setField(String field) {
+         this.field = field;
+      }
+
+      public List<String> getValue() {
+         return value;
+      }
+
+      public void setValue(List<String> value) {
+         this.value = value;
+      }
+
+      @Override
+      public String toString() {
+         final StringBuilder sb = new StringBuilder();
+         sb.append("Action");
+         sb.append("{field=").append(field);
+         sb.append(", value=").append(value);
+         sb.append('}');
+         return sb.toString();
+      }
+
+      // Zendesk sometimes returns JSON with a String instead of String[] so we need a custom deserializer
+      public static class ActionDeserializer extends JsonDeserializer<Action> {
```

Comment:
@artichikin we noticed the same thing but didn't manage to send a pull request with our implementation, the downside is it contains a pretty nasty hack for something we were not able to figure out..

```java
/**
 * @author Johno Crawford (johno@sulake.com)
 */
public class ActionDeserializer extends com.fasterxml.jackson.databind.JsonDeserializer<Action> {

    @Override
    public Action deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
        ObjectCodec objectCodec = jsonParser.getCodec();
        JsonNode root = objectCodec.readTree(jsonParser);
        String value = "";
        JsonNode valueNode = root.get("value");
        if (valueNode.isArray()) {
            for (JsonNode node : valueNode) {
                if (!node.asText().startsWith("channel:")) {
                    value = node.asText();
                    break;
                }
            }
        } else {
            value = valueNode.asText();
        }
        return new Action(root.get("field").asText(), value);
    }
}
```